core/rawdb/accessors_indexes.go

```diff
@@ -56,6 +57,20 @@ func WriteTxLookupEntries(db DatabaseWriter, block *types.Block) {
 			utils.Logger().Error().Err(err).Msg("Failed to store transaction lookup entry")
 		}
 	}
+	for i, tx := range block.StakingTransactions() {
+		entry := TxLookupEntry{
+			BlockHash:  block.Hash(),
+			BlockIndex: block.NumberU64(),
+			Index:      uint64(i),
```

Comment:
This will overlay with the normal transaction's lookup key. We should do len(Transactions)+i. Adding the index of staking txn on top of normal txns.